scripts/staging/slicing/sparked/sparked_utils.py

```diff
@@ -0,0 +1,122 @@
+from pyspark.sql.functions import udf
+from pyspark.sql.types import FloatType
+
+from slicing.base.SparkedNode import SparkedNode
+from slicing.base.slicer import opt_fun, union
+
+calc_loss = udf(lambda target, prediction, type: calc_loss_fun(target, prediction, type), FloatType())
+model_type_init = udf(lambda type: init_model_type(type))
+
+
+def calc_loss_fun(target, prediction, type):
+    if type == 0:
+        return (prediction - target) ** 2
+    elif type == 1:
+        if target == prediction:
+            return float(1)
+        else:
+            return float(0)
+
+
+def init_model_type(model_type):
+    if model_type == "regression":
+        return 0
+    elif model_type == "classification":
+        return 1
+
+
+def slice_join_nonsense(node_i, node_j, cur_lvl):
+    commons = 0
+    for attr1 in node_i.attributes:
+        for attr2 in node_j.attributes:
```

Comment:
Since the attributes are sorted, we can replace the nested loop join with a merge join that performs a sorted set intersection.